### PR TITLE
fix: #2 Pharmacy registration updated

### PR DESCRIPTION
**Pull request by milosmarkovic1701**
Hospital automatically sends its info to drugstore when drugstore registers itself in hospital, some code movement and refactoring.

### CODE REVIEW

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/HospitalController.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        public HospitalController(MyDbContext db) //Ovo mora da stoji, ne znam zasto!!!
+        {
+            this.dbContext = db;
+        }
+
+        [HttpPost]
+        public IActionResult Post(HospitalDto hospital)
+        {
+            hospitalService = new HospitalService(dbContext);
+            int id = hospitalService.GetMaxId();
```
Comment (lenche23) [inline src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/HospitalController.cs:38]: ne ikrementovati ručno id, postoji tip polja unutar baze koji to radi sam

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Service/HospitalService.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            HospitalRepository = new HospitalSqlRepository();
+            HospitalRepository.dbContext = dbContext;
+        }
+
+        public HospitalService()
+        {
+            HospitalRepository = new HospitalSqlRepository();
+        }
+
+        public int GetMaxId()
```
Comment (lenche23) [inline src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Service/HospitalService.cs:27]: nepotrebno

File: src/Integration API/Controllers/DrugstoreController.cs [hunk 3]
```diff
@@ -49,12 +50,38 @@ public IActionResult GetDrugstoreName(int id)
...
+            request.AddJsonBody(jsonBody);
+
+            IRestResponse response = client.Execute(request);
+
+            var content = response.Content; // {"message":" created."}
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                repo.dbContext = dbContext;
+                int maxId = repo.GetMaxId();
+                Drugstore ds = new Drugstore(++maxId, newPharmacy.DrugstoreName, newPharmacy.URLAddress, ApiKey, newPharmacy.Email, newPharmacy.Address);
```
Comment (lenche23) [inline src/Integration API/Controllers/DrugstoreController.cs:78]: ...

**Review by lenche23 — CHANGES_REQUESTED** [review]

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/HospitalController.cs (hunk 1 above)
Comment (milosmarkovic1701) [reply src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/HospitalController.cs:38]: MAX ID CE BITI SREDJEN !!!! Sto pre

**Review by lenche23 — APPROVED** [review]

**Pull request merged**